types/node/net.d.ts

```diff
@@ -18,7 +18,26 @@ declare module "net" {
         writable?: boolean;
     }
 
-    interface TcpSocketConnectOpts {
+    interface OnReadOpts {
+        buffer?: Uint8Array | (() => Uint8Array);
+        /**
+         * This function is called for every chunk of incoming data.
+         * Two arguments are passed to it: the number of bytes written to buffer and a reference to buffer.
+         * Return false from this function to implicitly pause() the socket.
+         */
+        callback?(bytesWritten: number, buf: Uint8Array): boolean;
```

Comment:
I think `callback` is mandatory here.

Reply:
I looked at the source and they don't seem mandatory.

Reply:
https://github.com/nodejs/node/blob/1447a79dc435578bffba32ead9b2ffa0425fc30a/lib/net.js#L295
Not 100% certain though, can you perhaps verify?

Reply:
Well, node allows to omit `buffer` and/or `callback` and falls back to normal operation like if `onRead` has not been given at all. But I don't think that there is any real use case for passing `onRead` but skip `callback` or `buffer`.

Reply:
Gotta love node code.. I will make them required :)